### PR TITLE
[#83] Pretty latex for function asymptotics

### LINKED ISSUE #83: Write pretty latex documentation for function asymptotics
Currently we can improve documentation for `Nub` module by using latex formatting:

* http://hackage.haskell.org/package/relude-0.3.0/docs/Relude-Nub.html

Maybe there're other places where we can improve docs.

### PR DESCRIPTION
**Pull request by achilleasNP**
Resolves #83 

### HLint

- [ ] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [ ] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall)
        accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
  - [ ] I've generated the new `.hlint.yaml` file (see 
        [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).

### General

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) 
      with the short description of my latest changes.
- [x] All new and existing tests pass.
- [ ] I keep the code style used in the files I've changed (see 
      [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) 
      for more details).
- [ ] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

File: src/Relude/Nub.hs [hunk 1]
```diff
@@ -58,7 +58,7 @@ ordNub = go Set.empty
       then go s xs
       else x : go (Set.insert x s) xs
 
-{- | Like 'Prelude.nub' but runs in @O(n * log_16(n))@ time and requires 'Hashable'.
+{- | Like 'Prelude.nub' but runs in \( O(n \log_{16} n) \)  time and requires 'Hashable'.
 
 >>> hashNub [3, 3, 3, 2, 2, -1, 1]
 [3,2,-1,1]
```
[the remark this answers is not in the document]
Comment (achilleasNP) [reply src/Relude/Nub.hs]: Sure, I will do that tomorrow.

[the remark this answers is not in the document]
Comment (achilleasNP) [reply src/Relude/Nub.hs]: Are you sure about the hashNub and unstableNub asymptotics. It seems to me that they depend on insert/member in HashSet and those depend on IntMap so they are of order O(min(n,W)) where W are are the bits of int representation (32 or 64), which would mean  that the asymptotics of hashNub and unstableNub are O( n min(n,W)).

BTW, I noticed that the module Relude.Nub  is using the type HashSet which seems to have been deprecated.

[the remark this answers is not in the document]
Comment (vrom911) [reply src/Relude/Nub.hs]: I guess we are thinking about different `HashSet`s. The one we are using here is from the `unordered-containers` package: 

* https://hackage.haskell.org/package/unordered-containers-0.2.9.0/docs/Data-HashSet.html 

And it's not deprecated. And that's why the asymptotics is different from what you're suggesting.
`HashSet` from the `hashmap` package is indeed deprecated:

* https://hackage.haskell.org/package/hashmap-1.3.3/docs/Data-HashSet.html

And it's literally just `IntMap`.

[the remark this answers is not in the document]
Comment (achilleasNP) [reply src/Relude/Nub.hs]: Thanks for the thorough response. It never even occurred to me that they are different types of HashSets so I learned something new.
I added asymptotics for the other two functions.

**Pull request merged**